Content.Shared/Alert/AlertsSystem.cs

```diff
@@ -94,8 +97,16 @@ public void ShowAlert(EntityUid euid, AlertType alertType, short? severity = nul
             // In the case we're changing the alert type but not the category, we need to remove it first.
             alertsComponent.Alerts.Remove(alert.AlertKey);
 
-            alertsComponent.Alerts[alert.AlertKey] = new AlertState
-                { Cooldown = cooldown, Severity = severity, Type = alertType };
+            var state = new AlertState
+                { Cooldown = cooldown, AutoRemove = autoRemove, Severity = severity, Type = alertType };
+
+            alertsComponent.Alerts[alert.AlertKey] = state;
+
+            if (alertsComponent.Alerts[alert.AlertKey].AutoRemove is not null )
+            {
+                var autoComp = EnsureComp<AlertAutoRemoveComponent>(euid);
+                autoComp.Alerts.Add(alert.AlertKey, state);
+            }
```

Comment:
What happens if the alert gets popped multiple times? Add would just error, should the timer be bumped.

Reply:
Fixed.
Turns out the reason this did not throw any errors in testing was because I failed to expand the comparison earlier in the code, and autoremove alerts were never updated, expiring at their originally set time then starting over on the next metabolism tick, instead of updating the autoremove time for as long as the alert kept getting triggered. Fixing this got rid of the blinking icon behaviour too, yay